### PR TITLE
Demo cleanup, commenting, refactor

### PR DESCRIPTION
**Pull request by lizadaily**
This PR contains some cleanup and edge case fixes I noticed while writing the third piece of content. It also contains a little bit of refactoring of the userMedia file to make it easier to write about (and hopefully read). Overview (I also left some comments in the actual code below to give a better idea of what's changing):

* More comments (partly used in content, thought it might be worth keeping them in the code as well)
* Removal of some unneeded imports
* More reliable zone leaving if moving from desk zone straight to broadcast zone
* If local user is moving to global zone, only try to set their texture back to video if their camera is enabled
* Remove unused parameter in `moveTo()` method.
* Remove some no longer needed calls to `getBounds()`
* If setting a default texture and current texture is video, destroy video texture
* Move audio graph to dedicated `NodeChain` class, keeping `UserMedia` less cluttered and making the graph logic more readable
* More reliable audio graph destruction

### CODE REVIEW

File: src/world.ts [hunk 1]
```diff
@@ -39,7 +39,54 @@ export class World {
   private furniture: Array<ICollider> = [];
 
   constructor() {
-    this.init();
+    const w = document.getElementById("world");
```
Comment (lizadaily) [inline src/world.ts:42]: This is just moved up from the `init()` function for smoother coverage in the content.

File: src/models/userMedia.ts [hunk 2]
```diff
@@ -23,28 +27,124 @@ export enum Action {
 const isChrome: boolean = !!(navigator.userAgent.indexOf("Chrome") !== -1);
 console.log("IsChrome", isChrome);
 
+// NodeChain is responsible for creating, destroying, and connecting our
+// Web Audio API nodes.
+class NodeChain {
```
Comment (lizadaily) [inline src/models/userMedia.ts:32]: This moves the audio graph creation and destruction into its own class from `UserMedia`, making it more readable and easier to cover in a straightforward way. It also handles proper disconnection of all the nodes.

File: src/world.ts [hunk 3]
```diff
@@ -189,36 +242,37 @@ export class World {
     this.app.ticker.add((deltaTime) => {
       this.update(deltaTime);
     });
+
+    // Ensure our world is correctly sized
     this.app.resize();
-    // Container that will hold our room "furniture" elements,
-    // like broadcast spots
-    this.furnitureContainer = new PIXI.Container();
```
Comment (lizadaily) [inline src/world.ts:195]: Moved this part into the constructor of the World class, to sit alongside the user container for consistency.

File: src/world.ts [hunk 4]
```diff
@@ -521,16 +553,4 @@ export class World {
     this.robots = [];
     this.app.destroy(true, true);
   }
-
-  private initAudioContext() {
-    window.audioContext = new AudioContext();
-    const listener = window.audioContext.listener;
```
Comment (lizadaily) [inline src/world.ts:527]: We are no longer using spatialized nodes like `PannerNode` (instead we control the pan directly using `StereoPannerNode`), so we don't actually need to do anything with the listener. Just removing this part.

**Review by jessmitch42 — APPROVED** [review]
LGTM! I did some basic manual testing and didn't notice any regressions.

**Pull request merged**